File: Coding/JavaScript/Set5/LuckNumber.js
function helper(num_position, counter){
    if(counter > num_position)
        return true
    if(num_position % counter == 0)
        return false
    return helper(num_position - (Math.floor(num_position/counter)), counter + 1)
}

function luckNumber(num){
    return helper(num, 2)
}
console.log(luckNumber(7))
console.log(luckNumber(9))
console.log(luckNumber(13))

File: Coding/Python/Set5/LuckNumber.py
def helper(num_pos, counter):
    if counter > num_pos:
        return True
    if num_pos % counter == 0:
        return False
    num_pos = num_pos - (num_pos // counter)
    return helper(num_pos, counter + 1)


def is_lucky_number(num):
    return helper(num, 2)


print(is_lucky_number(13))
print(is_lucky_number(5))
